doc/manual/noteworthy-differences.rst

```diff
@@ -251,3 +251,94 @@ Noteworthy differences from Python
   returns a new random number every time it is invoked without argument. On the
   other hand, the function ``g(x=[1,2]) = push!(x,3)`` returns ``[1,2,3]`` every
   time it is called as ``g()``.
+
+
+Noteworthy differences from C/C++
+----------------------------------
+
+- Julia arrays are indexed with square brackets, and can have more than one
+  dimension ``A[i,j]``.
+  This syntax is not just syntactic sugar for a reference to a pointer or address as in C/C++.
+  See the Julia documentation for the syntax for array construction (it has changed between versions).
+- In Julia, indexing of arrays, strings, etc. is 1-based not 0-based.
+- Julia arrays are assigned by reference. After ``A=B``, changing elements of
+  ``B`` will modify ``A`` as well.
+- Julia values are passed and assigned by reference. If a function modifies an
+  array, the changes will be visible in the caller.
+- In Julia, whitespace is significant, unlike C/C++, so care must be taken when adding/removing
+  whitespace from a Julia program.
```

Comment:
This seems unduly scary. There are not a lot of situations where whitespace is significant. I would tone down the "scare" aspect of this.